fontbe/src/name.rs

```diff
@@ -42,11 +48,25 @@ impl Work<Context, AnyWorkId, Error> for NameWork {
                 language_id: key.lang_id,
                 string: OffsetMarker::new(value.clone()),
             })
-            .collect::<Vec<_>>();
+            .collect();
+
+        if let Some(name) = extra_tables
+            .as_ref()
+            .and_then(|tables| tables.name.as_ref())
+        {
+            log::info!("merging name records from FEA");
+            merge_name_records(&mut name_records, name);
+        }
 
-        context
-            .name
-            .set(Name::new(name_records.into_iter().collect()));
+        context.name.set(Name::new(name_records));
         Ok(())
     }
 }
+
+fn merge_name_records(records: &mut BTreeSet<NameRecord>, fea_names: &Name) {
+    // what should we be doing, here?
+    // for now I am going to use the same logic as we have for handling manual
+    // GlyphClassDef statements in FEA, and assume that the FEA is correct?
+    // My confidence in this approach is medium.
```

Comment:
is that what fontmake does? - if so we should be more confident and ideally cite the fontmake code